agimus_controller/utils/build_models.py

```diff
@@ -132,38 +132,51 @@ def load_model(self, load_from_ros):
             urdf_string = rospy.get_param("robot_description")
             srdf_string = rospy.get_param("robot_description_semantic")
-
-        if not load_from_ros:
+            self.construct_robot_model(urdf_string, srdf_string, mesh_dir)
+            self.construct_collision_model(self.rmodel, urdf_string, yaml_path)
+        else:
             print("Load robot from files")
 
-            robot_package_path = Path(franka_description.__path__[0])
-            robot_dir_path = robot_package_path / "robots" / "panda"
+            urdf_path = str(get_project_root() / "urdf" / "robot.urdf")
+            srdf_path = str(get_project_root() / "srdf" / "demo.srdf")
+            robot = example_robot_data.load("panda")
+            self._rmodel = self.get_robot_model(robot, urdf_path, srdf_path)
+            self._crmodel = self.get_collision_model(urdf_path, yaml_path)
 
-            urdf_path = str(robot_dir_path / "panda.urdf.xacro")
-            srdf_path = str(robot_dir_path / "panda.srdf")
+    def get_robot_model(self, robot, urdf_path, srdf_path):
+        locked_joints = [
+            robot.model.getJointId("panda_finger_joint1"),
+            robot.model.getJointId("panda_finger_joint2"),
+        ]
 
-            urdf_string = process_xacro(urdf_path)
-            with open(srdf_path, "r") as f:
-                srdf_string = f.read()
+        model = pin.Model()
+        pin.buildModelFromUrdf(urdf_path, model)
+        pin.loadReferenceConfigurations(model, srdf_path, False)
+        q0 = model.referenceConfigurations["default"]
+        return pin.buildReducedModel(model, locked_joints, q0)
 
-        self.construct_robot_model(self.robot, urdf_string, srdf_string, mesh_dir)
-        self.construct_collision_model(self.rmodel, urdf_string, yaml_path)
+    def get_collision_model(self, urdf_path, yaml_path):
+        collision_model = pin.buildGeomFromUrdf(self._rmodel, urdf_path, pin.COLLISION)
+        collision_model = reduce_capsules_robot(collision_model)
+        parser = ObstacleParamsParser(yaml_path, collision_model)
+        parser.add_collisions()
+        return parser.collision_model
 
-    def construct_robot_model(self, robot, urdf_string, srdf_string, mesh_dir):
-        self._model, self._cmodel, self._vmodel = pin.buildModelFromXML(
-            urdf_string, mesh_dir
-        )
+    def construct_robot_model(self, urdf_string, srdf_string):
+        self._model = pin.buildModelFromXML(urdf_string)
 
         locked_joints = [
-            robot.model.getJointId("panda_finger_joint1"),
-            robot.model.getJointId("panda_finger_joint2"),
+            self._model.getJointId("panda_finger_joint1"),
+            self._model.getJointId("panda_finger_joint2"),
         ]
         pin.loadReferenceConfigurationsFromXML(self._model, srdf_string, False)
-        geom_models = [self._vmodel, self._cmodel]
-        q0 = self.model.referenceConfigurations["default"]
+        geom_model = pin.buildGeomFromUrdfString(
```

Comment:
should be called collision_model in this case then.
I would like to save in this class the complete models and the reduced models everytime.
Meaning `self._cmodel` and `self._crmodel` should exists.
I believe it would be nice for debugging purposes.